Mono.Debugging.Soft/SoftDebuggerSession.cs

```diff
@@ -1981,7 +1981,16 @@ void HandleAssemblyUnloadEvents (AssemblyUnloadEvent[] events)
 
 			if (assemblyLocation != null) {
 				foreach (var pair in source_to_type) {
-					pair.Value.RemoveAll (m => PathComparer.Equals (m.Assembly.Location, assemblyLocation));
+					pair.Value.RemoveAll (m => {
+						try {
+							assemblyLocation = m.Assembly.Location;
```

Comment:
What I mean is, this delegate will always return true, because we're setting assemblyLocation to m.Assembly.Location.